.github/workflows/build.yml

```diff
@@ -80,6 +80,31 @@ jobs:
         docker exec emscripten emcc -v
         docker exec emscripten emcmake cmake .
         docker exec emscripten make -j 2 VERBOSE=1
+  wasi:
+    name: wasi
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: true
+      - name: build-tools
+        run: |
+          docker run -di --name wasi-sdk -v $(pwd):/src --workdir /src ghcr.io/webassembly/wasi-sdk:wasi-sdk-20 bash
+          docker exec wasi-sdk cmake -S . -B out -G Ninja -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=0
+          # We only build tools, not runtime, for WASI build
+          docker exec wasi-sdk cmake --build out --config Release \
+          --target wast2json-copy-to-bin \
```

Comment:
This can be made simpler with edits to the cmake config, for an initial iteration I tried to be as least invasive as possible. But cmake could be tweaked to either

- Collect tools build, without library build, into one target
- Provide an option to exclude wasm-rt, which I think is the only target that doesn't build with wasi due to use of setjmp

I don't know if there's much use case for publishing the library as wasi so perhaps first option is simpler. Or can just stick with this approach that doesn't need any build config tweaks

Reply:
Yeah, lets figure out to make the `all` target build like it currently does for `emscripten`.    Perhaps that means adding an option to disable wasm-rt then lets do that.  (or we could do it based on detecting that `longjmp` exists?

Reply:
@sbc100 Thanks for the header check idea, that worked out quite well I think. One other tweak was link flags to libwasm.so, please verify the changes in cmakelists